src/main/java/digital/slovensko/autogram/server/dto/ValidationResponseBody.java

```diff
@@ -0,0 +1,158 @@
+                    d.getId(),
+                    MimeTypeEnum.PDF.getMimeTypeString(),
+                    d.getReferencedName()
+                )).toList();
+                break;
+            }
+            case XAdES: {
+                extractor = new ASiCWithXAdESContainerExtractor(document);
+                break;
+            }
+            case CAdES: {
+                extractor = new ASiCWithCAdESContainerExtractor(document);
+                break;
+            }
+            default:
+        }
+
+        if (extractor != null) {
+            signedObjects = getSignedObjects(extractor.extract().getSignedDocuments(), dd.getAllSignerDocuments());
+            unsignedObjects = getUnsignedObjects(extractor.extract().getSignedDocuments(), dd.getAllSignerDocuments());
+        }
+
+        return new ValidationResponseBody(fileFormat, signatures, signedObjects, unsignedObjects);
+    }
+
+    private static List<SignedObject> getSignedObjects(List<DSSDocument> docs, List<SignerDataWrapper> signedObjects) {
+        return signedObjects.stream().map((signedObject) -> {
+            var r = docs.stream().filter((doc) -> doc.getName().equals(signedObject.getReferencedName())).toList();
+            if (r.isEmpty())
+                return null;
+
+            return new SignedObject(
+                    signedObject.getId(),
+                    r.get(0).getMimeType().getMimeTypeString(),
+                    signedObject.getReferencedName()
+            );
+        }).toList();
+    }
+
+    private static List<UnsignedObject> getUnsignedObjects(List<DSSDocument> docs, List<SignerDataWrapper> signedObjects) {
+        return docs.stream().filter((o) -> signedObjects.stream().filter((s) -> o.getName().equals(s.getReferencedName())).toList().isEmpty()).map((generalObject) ->
+            new UnsignedObject(
+                    generalObject.getMimeType().getMimeTypeString(),
+                    generalObject.getName()
+            )
+        ).toList();
+    }
+}
+
+record Signature(Result validaitonResult, SignatureInfo signatureInfo) {
+}
+
+record Result(int code, String description) {
+}
+
+record SignatureInfo(String level, String claimedSigningTime, boolean isTimestamped, String timestampSigningTime,
+                     CertificateInfo signingCertificate, List<TimestampCertificateInfo> timestamps,
+                     List<String> signedObjectsIds) {
+}
+
+record CertificateInfo(String issuerDN, String subjectDN, String serialNumber, String productionTime, String notBefore,
+                       String notAfter, Result qualification) {
+}
+
+record TimestampCertificateInfo(String issuerDN, String subjectDN, String serialNumber, String productionTime, String notBefore,
+                       String notAfter, Result qualification, String timestampType) {
+}
+
+record SignedObject(String id, String mimeType, String filename) {
+}
+
+record UnsignedObject(String mimeType, String filename) {
```

Comment:
Zvykne sa toto davat do 1 suboru?

Reply:
Viem, že do istého času to Java na úrovni jazyka zakazovala. Časom ale zjavne existoval rozumný dôvod to povoliť. Podľa mňa to v tomto prípade dáva obrovský zmysel, inak by tu muselo byť zbytočne ďalších 6 súborov s recordami, ktoré nič nerobia a nikde inde sa nepoužívajú. Som názoru, že menej súborov je lepšie.

Reply:
Rozumiem, ale v tom pripade by som asi navrhol, ze nech to su records interne pre tu triedu, aj tak sa asi nikde inde nepouzivaju. 

Zatial sme vsade isli top level trieda = subor, tak to asi nahodne nemenme.

Reply:
dal som to aspoň do vnútra tej hlavnej classy.